include/arch/x86/arch/64/mode/fastpath/fastpath.h

```diff
@@ -203,7 +203,15 @@ static inline void NORETURN FORCE_INLINE fastpath_restore(word_t badge, word_t m
 #endif /* CONFIG_KERNEL_SKIM_WINDOW */
 #endif /* defined(ENABLE_SMP_SUPPORT) && defined(CONFIG_KERNEL_SKIM_WINDOW) */
             "sti\n"
-            "sysexitq\n"
+            /* Return to user.
+             *
+             * SYSEXIT  0F 35     ; Return to compatibility mode from fast system call.
+             * SYSEXITQ 48 0F 35  ; Return to 64-bit mode from fast system call.
+             *
+             * clang uses "sysexitq" and gcc uses "rex.w sysexit" to generate 0x48,0x0F,0x35
+             * only at gcc 12 or later will sysexitq work on both compilers
+             */
+            ".byte 0x48,0x0F,0x35 \n"
```

Comment:
We could put this in `include/arch/x86/arch/64/mode/hardware.h`
```
/*
 * SYSEXIT  0F 35     ; Return to compatibility mode from fast system call.
 * SYSEXITQ 48 0F 35  ; Return to 64-bit mode from fast system call.
 *
 * clang uses "sysexitq" and gcc uses "rex.w sysexit" to generate 0x48,0x0F,0x35
 * only at gcc 12 or later "sysexitq" works on both compilers.
 */
#if (defined(__clang__) || (defined(__GNUC__) && (__GNUC__ > 12)))
    #define SYSEXITQ "sysexitq"
#elif defined(__GNUC__)
    #define SYSEXITQ ".byte 0x48,0x0F,0x35"
#else
    #error "unsupported compiler"
#endif
```

and then just use this here (and also in src/arch/x86/64/c_traps.c);
```
            "sti\n"
            SYSEXITQ "\n" /* return to user */
            :
```

Reply:
That should at least allow the CParser to work without modifications

Reply:
Yes, I like that much better!